OpenRA.Mods.Common/Widgets/VideoPlayerWidget.cs

```diff
@@ -168,7 +170,7 @@ public override void Draw()
 				while (nextFrame > Video.CurrentFrameIndex)
 				{
 					Video.AdvanceFrame();
-					videoSprite.Sheet.GetTexture().SetData(Video.CurrentFrameData, textureSize, textureSize);
+					videoSprite.Sheet.GetTexture().SetData(Video.CurrentFrameData, textureWidth, textureHeight);
```

Comment:
skip SetData if you already know you need to advance multiple frames. 
i.e. assuming this would only happen on slow gpu hardware, best to skip the expensive set data's - as it will likely cause data to be sent to gpu memory.

Reply:
SetData() can be just move next to the log entry